backend/db/utils/user_utils.py

```diff
@@ -14,9 +30,10 @@ def user_exists(user) -> bool:
     with Session(engine) as session:
         existing_user = session.exec(select(Users).where(Users.user_id == user.user_id)).first()
         if not existing_user:
-            return False
+            return False, None
         else:
-            return True
+            last_date = get_last_email_date(user.user_id)
+            return True, last_date
```

Comment:
```suggestion
            return True, last_fetched_date
```
Maybe we can rename for clarity